src/discord/discord.service.ts

```diff
@@ -21,6 +34,24 @@ export class DiscordService implements OnModuleInit {
         await interaction.reply('Pong!');
       }
     });
+
+    this.client.on('messageCreate', (msg) => {
+      if (msg.author.bot) return;
+
+      const isFromTestChannel =
+        msg.channelId ===
+        this.configService.getOrThrow('DISCORD_BOT_TESTING_CHANNEL_ID');
+      const isProduction = this.configService.get('NODE_ENV') === 'production';
+
+      if (msg.channel.type === 0 && (isFromTestChannel || isProduction)) {
```

Comment:
Could you use `ChannelType.GuildText` instead of zero? `0` doesn't convey the intention very well.

Found some info here: https://discord-api-types.dev/api/discord-api-types-v10/enum/ChannelType

Reply:
What do you think about `msg.channel instanceof TextChannel` and `msg.channel instanceof DMChannel`?
https://alon-l.github.io/node-discord/interfaces/textchannel.html

Reply:
That sounds good!